docs/source/design/python_imports.rst

```diff
@@ -0,0 +1,65 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_python_imports:
+
+************************
+Python import statements
+************************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current (albeit incomplete) implementation.
+
+The following rules apply to any ``import`` statement in the code base:
+
+- All imports *must* be absolute, unless they import individual pieces of an integrated code component that is only split across several source code files for technical or organizational reasons.
```

Comment:
Such description reminded me about my first "large" program in Borland Pascal 7.0 on MS-DOS in 1995 where I had to learn about modularization via multiple files because the whole thing could no longer fit into maximal allowed per file 64KB ;)

But I think it is too vague and non-specific, thus provides no "rule" really. The entire `datalad` package is split into submodules for "organizational" reasons (if not at times technical such as "import time"), but we do not want to use relative imports through out.

Reply:
If you have a concrete suggestion on how to improve on this wording without turning it into a full page, I'd be happy to discuss.

I do not agree with the "there is no rule" judgement, though. The rule is that all imports must be absolute, unless a developer has a good reason to make an exception.

I am fairly confident that developing an explicit list of exceptions is a waste of time in a community of a handful developers, and that no list will spare us a conversation about import styles in the future. What I propose is a cheap compromise between "must be absolute", and the demands/rationals that you outlined earlier.

Reply:
> I do not agree with the "there is no rule" judgement, though. The rule is that all imports must be absolute, unless a developer has a good reason to make an exception.

well, one developer decides it is worth exception (e.g. me on tests and "backends" etc imports) and another decides the other way -- and hence it is where we have started from really.  But I agree that this is must better than nothing due to other information in the document etc.  So indeed worthwhile proceeding as is for now.

Reply:
:+1: